src/vos/vos_io.c

```diff
@@ -1659,6 +1661,71 @@ akey_update_recx(daos_handle_t toh, uint32_t pm_ver, daos_recx_t *recx,
 	return rc;
 }
 
+int
+vos_key_flags_set(struct umem_instance *umm, struct vos_krec_df *krec)
+{
+	int	rc;
+	uint8_t	bmap;
+
+	bmap = krec->kr_bmap;
+	if ((bmap & KREC_BF_AGG_OPT) == 0)
+		return 0;
+
+	if ((bmap & KREC_BF_AGG_NEEDED) == 0)
+		bmap |= KREC_BF_AGG_NEEDED;
```

Comment:
I don't see why the KREC_BF_AGG_FLAG isn't cleared in such case.

Reply:
It should never be set in such case.

Reply:
oh, right. probably adding an assert here?